library/network_connections.py

```diff
@@ -2012,7 +2012,12 @@ def run_prepare(self):
                                 "interface exists" % (connection["interface_name"]),
                             )
                         elif connection["type"] == "infiniband":
-                            if connection["infiniband"]["p_key"] != -1:
+                            # The default `p_key` value is -1, which means that the
+                            # `p_key` is unset. In this case, the physical infiniband
+                            # device is required to be present for creating the
+                            # connection on it
+                            default_pkey_val = -1
+                            if connection["infiniband"]["p_key"] == default_pkey_val:
```

Comment:
IMHO the constant should be used in all place where `-1` is used for `p_key`, so also in the place where the default value is assigned. This is to make sure that it is consistently used everywhere.

Reply:
I updated, thanks.